newIDE/app/src/AssetStore/PrivateAssets/PrivateAssetPackPurchaseDialog.js

```diff
@@ -44,14 +83,22 @@ const PrivateAssetPackPurchaseDialog = ({
     setIsCheckingPurchasesAfterLogin,
   ] = React.useState(!receivedAssetPacks);
   const [purchaseSuccessful, setPurchaseSuccessful] = React.useState(false);
+  const [
+    displayPasswordPrompt,
+    setDisplayPasswordPrompt,
+  ] = React.useState<boolean>(false);
+  const [password, setPassword] = React.useState<string>('');
+
   const onStartPurchase = async () => {
     if (!profile) return;
+    setDisplayPasswordPrompt(false);
     try {
       setIsPurchasing(true);
       const checkoutUrl = await getStripeCheckoutUrl(getAuthorizationHeader, {
         stripePriceId: privateAssetPackListingData.prices[0].stripePriceId,
         userId: profile.id,
         customerEmail: profile.email,
+        ...(password ? { password } : undefined),
       });
       Window.openExternalURL(checkoutUrl);
     } catch (error) {
```

Comment:
let's read the error status here and display a "Unauthorized" if that's the case?